diff -r 666a44dfa6eb build.gradle
--- a/build.gradle      Thu Jan 17 21:09:04 2019 +0000
+++ b/build.gradle      Mon Jan 21 00:00:00 2019 +0000
@@ -27,10 +27,11 @@
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-            }
+        maven {
+            url "https://maven.google.com/"
+        }
+        maven {
+            url "https://jcenter.bintray.com/"
         }
     }
 }
@@ -39,10 +40,11 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-            }
+        maven {
+            url "https://maven.google.com/"
+        }
+        maven {
+            url "https://plugins.gradle.org/m2/"
         }
     }
 
diff -r 666a44dfa6eb mobile/android/geckoview/build.gradle
--- a/mobile/android/geckoview/build.gradle     Thu Jan 17 21:09:04 2019 +0000
+++ b/mobile/android/geckoview/build.gradle     Mon Jan 21 00:00:00 2019 +0000
@@ -350,9 +350,6 @@
         }
     }
     repositories {
-        maven {
-            url = "${project.buildDir}/maven"
-        }
     }
 }
 
