diff -r 98d536130349 python/mozbuild/mozbuild/frontend/emitter.py
--- a/python/mozbuild/mozbuild/frontend/emitter.py	Wed Apr 18 12:17:09 2018 -0600
+++ b/python/mozbuild/mozbuild/frontend/emitter.py	Sun Apr 22 00:00:00 2018 +0000
@@ -1159,6 +1159,9 @@
             all_files = context.get(var)
             if not all_files:
                 continue
+            if var == 'TEST_HARNESS_FILES':
+                print("IGNORING TEST FILES")
+                continue
             if dist_install is False and var != 'TEST_HARNESS_FILES':
                 raise SandboxValidationError(
                     '%s cannot be used with DIST_INSTALL = False' % var,
@@ -1460,11 +1463,11 @@
                             not os.path.isfile(mozpath.join(context.config.topsrcdir,
                                                             install_path[2:])),
                             install_path not in install_info.external_installs]):
-                        raise SandboxValidationError('Error processing test '
+                        print('IGNORED: Error processing test '
                            'manifest %s: entry in support-files not present '
-                           'in the srcdir: %s' % (path, install_path), context)
-
-                obj.deferred_installs |= install_info.deferred_installs
+                           'in the srcdir: %s' % (path, install_path))
+                    else:
+                        obj.deferred_installs.add(install_path)
 
             for test in filtered:
                 obj.tests.append(test)
diff -r 98d536130349 python/mozbuild/mozbuild/frontend/reader.py
--- a/python/mozbuild/mozbuild/frontend/reader.py	Wed Apr 18 12:17:09 2018 -0600
+++ b/python/mozbuild/mozbuild/frontend/reader.py	Sun Apr 22 00:00:00 2018 +0000
@@ -1057,6 +1057,10 @@
                                          metadata=metadata):
                 yield s
 
+        except IOError:
+            print('IGNORING MISSING MOZBUILD %s' % path)
+            return
+
         except BuildReaderError as bre:
             raise bre
 
diff -r 98d536130349 python/mozbuild/mozbuild/frontend/sandbox.py
--- a/python/mozbuild/mozbuild/frontend/sandbox.py	Wed Apr 18 12:17:09 2018 -0600
+++ b/python/mozbuild/mozbuild/frontend/sandbox.py	Sun Apr 22 00:00:00 2018 +0000
@@ -153,6 +153,8 @@
 
         try:
             source = self._finder.get(path).read()
+        except AttributeError:
+            raise IOError()
         except Exception as e:
             raise SandboxLoadError(self._context.source_stack,
                 sys.exc_info()[2], read_error=path)
diff -r 98d536130349 python/mozbuild/mozbuild/util.py
--- a/python/mozbuild/mozbuild/util.py	Wed Apr 18 12:17:09 2018 -0600
+++ b/python/mozbuild/mozbuild/util.py	Sun Apr 22 00:00:00 2018 +0000
@@ -526,8 +526,14 @@
         return super(ListWithActionMixin, self).__setslice__(i, j, sequence)
 
     def __iadd__(self, other):
-        other = [self._action(i) for i in other]
-        return super(ListWithActionMixin, self).__iadd__(other)
+        other_new = []
+        for i in other:
+            try:
+                new = self._action(i)
+                other_new.append(new)
+            except IOError as e:
+                print('IGNORED:', e)
+        return super(ListWithActionMixin, self).__iadd__(other_new)
 
 class StrictOrderingOnAppendListWithAction(StrictOrderingOnAppendListMixin,
     ListMixin, ListWithActionMixin, list):
