diff -r 5e7156ebdc78 mobile/android/base/java/org/mozilla/gecko/BrowserApp.java
--- a/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java	Wed Jan 20 18:40:08 2016 -0500
+++ b/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java	Wed Jan 20 22:12:02 2016 -0300
@@ -28,8 +28,6 @@
 import org.mozilla.gecko.gfx.DynamicToolbarAnimator;
 import org.mozilla.gecko.gfx.ImmutableViewportMetrics;
 import org.mozilla.gecko.gfx.LayerView;
-import org.mozilla.gecko.health.HealthRecorder;
-import org.mozilla.gecko.health.SessionInformation;
 import org.mozilla.gecko.home.BrowserSearch;
 import org.mozilla.gecko.home.HomeBanner;
 import org.mozilla.gecko.home.HomeConfig.PanelType;
@@ -2275,16 +2273,7 @@
      *        {@link BrowserHealthRecorder#SEARCH_LOCATIONS}.
      */
     private static void recordSearch(SearchEngine engine, String where) {
-        //try {
-        //    String identifier = (engine == null) ? "other" : engine.getEngineIdentifier();
-        //    JSONObject message = new JSONObject();
-        //    message.put("type", BrowserHealthRecorder.EVENT_SEARCH);
-        //    message.put("location", where);
-        //    message.put("identifier", identifier);
-        //    EventDispatcher.getInstance().dispatchEvent(message, null);
-        //} catch (Exception e) {
-        //    Log.e(LOGTAG, "Error recording search.", e);
-        //}
+        return;
     }
 
     /**
diff -r 5e7156ebdc78 mobile/android/base/java/org/mozilla/gecko/GeckoApp.java
--- a/mobile/android/base/java/org/mozilla/gecko/GeckoApp.java	Wed Jan 20 18:40:08 2016 -0500
+++ b/mobile/android/base/java/org/mozilla/gecko/GeckoApp.java	Wed Jan 20 22:12:02 2016 -0300
@@ -15,9 +15,6 @@
 import org.mozilla.gecko.gfx.Layer;
 import org.mozilla.gecko.gfx.LayerView;
 import org.mozilla.gecko.gfx.PluginLayer;
-import org.mozilla.gecko.health.HealthRecorder;
-import org.mozilla.gecko.health.SessionInformation;
-import org.mozilla.gecko.health.StubbedHealthRecorder;
 import org.mozilla.gecko.menu.GeckoMenu;
 import org.mozilla.gecko.menu.GeckoMenuInflater;
 import org.mozilla.gecko.menu.MenuPanel;
@@ -207,7 +204,6 @@
 
     private String mPrivateBrowsingSession;
 
-    private volatile HealthRecorder mHealthRecorder;
     private volatile Locale mLastLocale;
 
     private EventListener mWebappEventListener;
@@ -703,15 +699,6 @@
                 mGeckoReadyStartupTimer.stop();
                 geckoConnected();
 
-                // This method is already running on the background thread, so we
-                // know that mHealthRecorder will exist. That doesn't stop us being
-                // paranoid.
-                // This method is cheap, so don't spawn a new runnable.
-                final HealthRecorder rec = mHealthRecorder;
-                if (rec != null) {
-                  rec.recordGeckoStartupTime(mGeckoReadyStartupTimer.getElapsed());
-                }
-
             } else if (event.equals("Gecko:Exited")) {
                 // Gecko thread exited first; let GeckoApp die too.
                 doShutdown();
@@ -1420,11 +1407,6 @@
                 final LocaleManager localeManager = BrowserLocaleManager.getInstance();
                 localeManager.initialize(getApplicationContext());
 
-                SessionInformation previousSession = SessionInformation.fromSharedPrefs(prefs);
-                if (previousSession.wasKilled()) {
-                    Telemetry.addToHistogram("FENNEC_WAS_KILLED", 1);
-                }
-
                 SharedPreferences.Editor editor = prefs.edit();
                 editor.putBoolean(GeckoApp.PREFS_OOM_EXCEPTION, false);
 
@@ -1452,13 +1434,6 @@
                     appLocaleString = osLocaleString;
                 }
 
-                mHealthRecorder = GeckoApp.this.createHealthRecorder(GeckoApp.this,
-                                                                     profilePath,
-                                                                     dispatcher,
-                                                                     osLocaleString,
-                                                                     appLocaleString,
-                                                                     previousSession);
-
                 final String uiLocale = appLocaleString;
                 ThreadUtils.postToUiThread(new Runnable() {
                     @Override
@@ -1672,34 +1647,6 @@
                                            (TextSelectionHandle) findViewById(R.id.caret_handle),
                                            (TextSelectionHandle) findViewById(R.id.focus_handle));
 
-        // Trigger the completion of the telemetry timer that wraps activity startup,
-        // then grab the duration to give to FHR.
-        mJavaUiStartupTimer.stop();
-        final long javaDuration = mJavaUiStartupTimer.getElapsed();
-
-        ThreadUtils.getBackgroundHandler().postDelayed(new Runnable() {
-            @Override
-            public void run() {
-                final HealthRecorder rec = mHealthRecorder;
-                if (rec != null) {
-                    rec.recordJavaStartupTime(javaDuration);
-                }
-
-                // Kick off our background services. We do this by invoking the broadcast
-                // receiver, which uses the system alarm infrastructure to perform tasks at
-                // intervals.
-                GeckoPreferences.broadcastHealthReportUploadPref(GeckoApp.this);
-            }
-        }, 50);
-
-        final int updateServiceDelay = 30 * 1000;
-        ThreadUtils.getBackgroundHandler().postDelayed(new Runnable() {
-            @Override
-            public void run() {
-                UpdateServiceHelper.registerForUpdates(GeckoApp.this);
-            }
-        }, updateServiceDelay);
-
         if (mIsRestoringActivity) {
             Tab selectedTab = Tabs.getInstance().getSelectedTab();
             if (selectedTab != null) {
@@ -2039,29 +1986,6 @@
         final long now = System.currentTimeMillis();
         final long realTime = android.os.SystemClock.elapsedRealtime();
 
-        ThreadUtils.postToBackgroundThread(new Runnable() {
-            @Override
-            public void run() {
-                // Now construct the new session on HealthRecorder's behalf. We do this here
-                // so it can benefit from a single near-startup prefs commit.
-                SessionInformation currentSession = new SessionInformation(now, realTime);
-
-                SharedPreferences prefs = GeckoApp.this.getSharedPreferences();
-                SharedPreferences.Editor editor = prefs.edit();
-                editor.putBoolean(GeckoApp.PREFS_WAS_STOPPED, false);
-                currentSession.recordBegin(editor);
-                editor.apply();
-
-                final HealthRecorder rec = mHealthRecorder;
-                if (rec != null) {
-                    rec.setCurrentSession(currentSession);
-                    rec.processDelayed();
-                } else {
-                    Log.w(LOGTAG, "Can't record session: rec is null.");
-                }
-            }
-        });
-
         Restrictions.update(this);
     }
 
@@ -2084,7 +2008,6 @@
     @Override
     public void onPause()
     {
-        final HealthRecorder rec = mHealthRecorder;
         final Context context = this;
 
         // In some way it's sad that Android will trigger StrictMode warnings
@@ -2096,9 +2019,6 @@
                 SharedPreferences prefs = GeckoApp.this.getSharedPreferences();
                 SharedPreferences.Editor editor = prefs.edit();
                 editor.putBoolean(GeckoApp.PREFS_WAS_STOPPED, true);
-                if (rec != null) {
-                    rec.recordSessionEnd("P", editor);
-                }
 
                 // If we haven't done it before, cleanup any old files in our old temp dir
                 if (prefs.getBoolean(GeckoApp.PREFS_CLEANUP_TEMP_FILES, true)) {
@@ -2109,11 +2029,6 @@
                 }
 
                 editor.apply();
-
-                // In theory, the first browser session will not run long enough that we need to
-                // prune during it and we'd rather run it when the browser is inactive so we wait
-                // until here to register the prune service.
-                GeckoPreferences.broadcastHealthReportPrune(context);
             }
         });
 
@@ -2212,18 +2127,6 @@
             }
         }
 
-        final HealthRecorder rec = mHealthRecorder;
-        mHealthRecorder = null;
-        if (rec != null && rec.isEnabled()) {
-            // Closing a HealthRecorder could incur a write.
-            ThreadUtils.postToBackgroundThread(new Runnable() {
-                @Override
-                public void run() {
-                    rec.close(GeckoApp.this);
-                }
-            });
-        }
-
         Favicons.close();
 
         super.onDestroy();
@@ -2777,15 +2680,6 @@
         final boolean startNewSession = true;
         final boolean shouldRestart = false;
 
-        // If the HealthRecorder is not yet initialized (unlikely), the locale change won't
-        // trigger a session transition and subsequent events will be recorded in an environment
-        // with the wrong locale.
-        final HealthRecorder rec = mHealthRecorder;
-        if (rec != null) {
-            rec.onAppLocaleChanged(locale);
-            rec.onEnvironmentChanged(startNewSession, SESSION_END_LOCALE_CHANGED);
-        }
-
         if (!shouldRestart) {
             ThreadUtils.postToUiThread(new Runnable() {
                 @Override
@@ -2840,16 +2734,6 @@
         });
     }
 
-    protected HealthRecorder createHealthRecorder(final Context context,
-                                                  final String profilePath,
-                                                  final EventDispatcher dispatcher,
-                                                  final String osLocale,
-                                                  final String appLocale,
-                                                  final SessionInformation previousSession) {
-        // GeckoApp does not need to record any health information - return a stub.
-        return new StubbedHealthRecorder();
-    }
-
     protected StartupAction getStartupAction(final String passedURL) {
         // Default to NORMAL here. Subclasses can handle the other types.
         return StartupAction.NORMAL;
diff -r 5e7156ebdc78 mobile/android/base/android-services.mozbuild
--- a/mobile/android/base/android-services.mozbuild    Wed Jan 20 18:40:08 2016 -0500
+++ b/mobile/android/base/android-services.mozbuild    Wed Jan 20 22:12:02 2016 -0300
@@ -780,7 +780,6 @@
     'background/common/log/writers/ThreadLocalTagLogWriter.java',
     'background/common/PrefsBranch.java',
     'background/common/telemetry/TelemetryWrapper.java',
-    'background/datareporting/TelemetryRecorder.java',
     'background/db/CursorDumper.java',
     'background/db/Tab.java',
     'background/fxa/FxAccount10AuthDelegate.java',
diff -r 5e7156ebdc78 mobile/android/base/moz.build
--- a/mobile/android/base/moz.build	Wed Jan 20 18:40:08 2016 -0500
+++ b/mobile/android/base/moz.build	Wed Jan 20 22:12:02 2016 -0300
@@ -325,9 +325,6 @@
     'gfx/VirtualLayer.java',
     'GlobalHistory.java',
     'GuestSession.java',
-    'health/HealthRecorder.java',
-    'health/SessionInformation.java',
-    'health/StubbedHealthRecorder.java',
     'home/BookmarkFolderView.java',
     'home/BookmarksListAdapter.java',
     'home/BookmarksListView.java',
diff -r 5e7156ebdc78 mobile/android/base/java/org/mozilla/gecko/preferences/GeckoPreferences.java
--- a/mobile/android/base/java/org/mozilla/gecko/preferences/GeckoPreferences.java	Wed Jan 20 18:40:08 2016 -0500
+++ b/mobile/android/base/java/org/mozilla/gecko/preferences/GeckoPreferences.java	Wed Jan 20 22:12:02 2016 -0300
@@ -1035,31 +1035,6 @@
 
     /**
      * Broadcast the provided value as the value of the
-     * <code>PREFS_HEALTHREPORT_UPLOAD_ENABLED</code> pref.
-     */
-    public static void broadcastHealthReportUploadPref(final Context context, final boolean value) {
-        //broadcastPrefAction(context,
-        //                    HealthReportConstants.ACTION_HEALTHREPORT_UPLOAD_PREF,
-        //                    PREFS_HEALTHREPORT_UPLOAD_ENABLED,
-        //                    value);
-    }
-
-    /**
-     * Broadcast the current value of the
-     * <code>PREFS_HEALTHREPORT_UPLOAD_ENABLED</code> pref.
-     */
-    public static void broadcastHealthReportUploadPref(final Context context) {
-        //final boolean value = getBooleanPref(context, PREFS_HEALTHREPORT_UPLOAD_ENABLED, true);
-        //broadcastHealthReportUploadPref(context, value);
-    }
-
-    public static void broadcastHealthReportPrune(final Context context) {
-        //final Intent intent = new Intent(HealthReportConstants.ACTION_HEALTHREPORT_PRUNE);
-        //broadcastAction(context, intent);
-    }
-
-    /**
-     * Broadcast the provided value as the value of the
      * <code>PREFS_GEO_REPORTING</code> pref.
      */
     public static void broadcastStumblerPref(final Context context, final boolean value) {
@@ -1245,14 +1220,6 @@
             UpdateServiceHelper.setAutoDownloadPolicy(this, UpdateService.AutoDownloadPolicy.get((String) newValue));
         } else if (PREFS_UPDATER_URL.equals(prefName)) {
             UpdateServiceHelper.setUpdateUrl(this, (String) newValue);
-        } else if (PREFS_HEALTHREPORT_UPLOAD_ENABLED.equals(prefName)) {
-            // The healthreport pref only lives in Android, so we do not persist
-            // to Gecko, but we do broadcast intent to the health report
-            // background uploader service, which will start or stop the
-            // repeated background upload attempts.
-            final Boolean newBooleanValue = (Boolean) newValue;
-            broadcastHealthReportUploadPref(this, newBooleanValue);
-            AdjustConstants.getAdjustHelper().setEnabled(newBooleanValue);
         } else if (PREFS_GEO_REPORTING.equals(prefName)) {
             broadcastStumblerPref(this, (Boolean) newValue);
             // Translate boolean value to int for geo reporting pref.
diff -r 5e7156ebdc78 mobile/android/search/java/org/mozilla/search/SearchActivity.java
--- a/mobile/android/search/java/org/mozilla/search/SearchActivity.java	Wed Jan 20 18:40:08 2016 -0500
+++ b/mobile/android/search/java/org/mozilla/search/SearchActivity.java	Wed Jan 20 22:12:02 2016 -0300
@@ -253,14 +253,6 @@
     public void onSearch(String query, SuggestionAnimation suggestionAnimation) {
         storeQuery(query);
 
-        try {
-            //BrowserHealthRecorder.recordSearchDelayed("activity", engine.getIdentifier());
-        } catch (Exception e) {
-            // This should never happen: it'll only throw if the
-            // search location is wrong. But let's not tempt fate.
-            Log.w(LOGTAG, "Unable to record search.");
-        }
-
         startSearch(query);
 
         if (suggestionAnimation != null) {
