diff -r 78278c4dc730 mobile/android/base/java/org/mozilla/gecko/GeckoApp.java
--- a/mobile/android/base/java/org/mozilla/gecko/GeckoApp.java	Wed Aug 02 14:04:05 2017 -0400
+++ b/mobile/android/base/java/org/mozilla/gecko/GeckoApp.java	Thu Aug 03 00:00:00 2017 -0000
@@ -10,9 +10,6 @@
 import org.mozilla.gecko.GeckoProfileDirectories.NoMozillaDirectoryException;
 import org.mozilla.gecko.annotation.RobocopTarget;
 import org.mozilla.gecko.annotation.WrapForJNI;
-import org.mozilla.gecko.health.HealthRecorder;
-import org.mozilla.gecko.health.SessionInformation;
-import org.mozilla.gecko.health.StubbedHealthRecorder;
 import org.mozilla.gecko.home.HomeConfig.PanelType;
 import org.mozilla.gecko.menu.GeckoMenu;
 import org.mozilla.gecko.menu.GeckoMenuInflater;
@@ -362,7 +359,6 @@
 
     private String mPrivateBrowsingSession;
 
-    private volatile HealthRecorder mHealthRecorder;
     private volatile Locale mLastLocale;
 
     private boolean mShutdownOnDestroy;
@@ -708,15 +704,6 @@
             mGeckoReadyStartupTimer.stop();
             geckoConnected();
 
-            // This method is already running on the background thread, so we
-            // know that mHealthRecorder will exist. That doesn't stop us being
-            // paranoid.
-            // This method is cheap, so don't spawn a new runnable.
-            final HealthRecorder rec = mHealthRecorder;
-            if (rec != null) {
-              rec.recordGeckoStartupTime(mGeckoReadyStartupTimer.getElapsed());
-            }
-
             ((GeckoApplication) getApplicationContext()).onDelayedStartup();
 
             // Reset the crash loop counter if we remain alive for at least half a minute.
@@ -1523,11 +1510,6 @@
                 final LocaleManager localeManager = BrowserLocaleManager.getInstance();
                 localeManager.initialize(getApplicationContext());
 
-                SessionInformation previousSession = SessionInformation.fromSharedPrefs(prefs);
-                if (previousSession.wasKilled()) {
-                    Telemetry.addToHistogram("FENNEC_WAS_KILLED", 1);
-                }
-
                 SharedPreferences.Editor editor = prefs.edit();
                 editor.putBoolean(GeckoAppShell.PREFS_OOM_EXCEPTION, false);
 
@@ -1555,13 +1537,6 @@
                     appLocaleString = osLocaleString;
                 }
 
-                mHealthRecorder = GeckoApp.this.createHealthRecorder(GeckoApp.this,
-                                                                     profilePath,
-                                                                     dispatcher,
-                                                                     osLocaleString,
-                                                                     appLocaleString,
-                                                                     previousSession);
-
                 final String uiLocale = appLocaleString;
                 ThreadUtils.postToUiThread(new Runnable() {
                     @Override
@@ -1835,29 +1810,6 @@
 
         mPromptService = new PromptService(this, getAppEventDispatcher());
 
-        // Trigger the completion of the telemetry timer that wraps activity startup,
-        // then grab the duration to give to FHR.
-        mJavaUiStartupTimer.stop();
-        final long javaDuration = mJavaUiStartupTimer.getElapsed();
-
-        ThreadUtils.getBackgroundHandler().postDelayed(new Runnable() {
-            @Override
-            public void run() {
-                final HealthRecorder rec = mHealthRecorder;
-                if (rec != null) {
-                    rec.recordJavaStartupTime(javaDuration);
-                }
-            }
-        }, 50);
-
-        final int updateServiceDelay = 30 * 1000;
-        ThreadUtils.getBackgroundHandler().postDelayed(new Runnable() {
-            @Override
-            public void run() {
-                UpdateServiceHelper.registerForUpdates(GeckoAppShell.getApplicationContext());
-            }
-        }, updateServiceDelay);
-
         if (mIsRestoringActivity) {
             Tab selectedTab = Tabs.getInstance().getSelectedTab();
             if (selectedTab != null) {
@@ -2301,36 +2253,6 @@
         final long now = System.currentTimeMillis();
         final long realTime = android.os.SystemClock.elapsedRealtime();
 
-        ThreadUtils.postToBackgroundThread(new Runnable() {
-            @Override
-            public void run() {
-                // Now construct the new session on HealthRecorder's behalf. We do this here
-                // so it can benefit from a single near-startup prefs commit.
-                SessionInformation currentSession = new SessionInformation(now, realTime);
-
-                SharedPreferences prefs = GeckoApp.this.getSharedPreferences();
-                SharedPreferences.Editor editor = prefs.edit();
-                editor.putBoolean(GeckoApp.PREFS_WAS_STOPPED, false);
-
-                if (!mLastSessionCrashed) {
-                    // The last session terminated normally,
-                    // so we can reset the count of successive crashes.
-                    editor.putInt(GeckoApp.PREFS_CRASHED_COUNT, 0);
-                }
-
-                currentSession.recordBegin(editor);
-                editor.apply();
-
-                final HealthRecorder rec = mHealthRecorder;
-                if (rec != null) {
-                    rec.setCurrentSession(currentSession);
-                    rec.processDelayed();
-                } else {
-                    Log.w(LOGTAG, "Can't record session: rec is null.");
-                }
-            }
-        });
-
         Restrictions.update(this);
     }
 
@@ -2370,7 +2292,6 @@
 
         mLastActiveGeckoApp = new WeakReference<GeckoApp>(this);
 
-        final HealthRecorder rec = mHealthRecorder;
         final Context context = this;
 
         // In some way it's sad that Android will trigger StrictMode warnings
@@ -2382,9 +2303,6 @@
                 SharedPreferences prefs = GeckoApp.this.getSharedPreferences();
                 SharedPreferences.Editor editor = prefs.edit();
                 editor.putBoolean(GeckoApp.PREFS_WAS_STOPPED, true);
-                if (rec != null) {
-                    rec.recordSessionEnd("P", editor);
-                }
 
                 // onPause might in fact be called even after a crash, but in that case the
                 // crash reporter will record this fact for us and we'll pick it up in onCreate.
@@ -2479,18 +2397,6 @@
             mPromptService = null;
         }
 
-        final HealthRecorder rec = mHealthRecorder;
-        mHealthRecorder = null;
-        if (rec != null && rec.isEnabled()) {
-            // Closing a HealthRecorder could incur a write.
-            ThreadUtils.postToBackgroundThread(new Runnable() {
-                @Override
-                public void run() {
-                    rec.close(GeckoApp.this);
-                }
-            });
-        }
-
         super.onDestroy();
 
         Tabs.unregisterOnTabsChangedListener(this);
@@ -2879,15 +2785,6 @@
         final boolean startNewSession = true;
         final boolean shouldRestart = false;
 
-        // If the HealthRecorder is not yet initialized (unlikely), the locale change won't
-        // trigger a session transition and subsequent events will be recorded in an environment
-        // with the wrong locale.
-        final HealthRecorder rec = mHealthRecorder;
-        if (rec != null) {
-            rec.onAppLocaleChanged(locale);
-            rec.onEnvironmentChanged(startNewSession, SESSION_END_LOCALE_CHANGED);
-        }
-
         final Runnable runnable = new Runnable() {
             @Override
             public void run() {
@@ -2929,16 +2826,6 @@
         onLocaleChanged(resultant);
     }
 
-    protected HealthRecorder createHealthRecorder(final Context context,
-                                                  final String profilePath,
-                                                  final EventDispatcher dispatcher,
-                                                  final String osLocale,
-                                                  final String appLocale,
-                                                  final SessionInformation previousSession) {
-        // GeckoApp does not need to record any health information - return a stub.
-        return new StubbedHealthRecorder();
-    }
-
     protected void recordStartupActionTelemetry(final String passedURL, final String action) {
     }
 
diff -r 78278c4dc730 mobile/android/base/java/org/mozilla/gecko/preferences/GeckoPreferences.java
--- a/mobile/android/base/java/org/mozilla/gecko/preferences/GeckoPreferences.java	Wed Aug 02 14:04:05 2017 -0400
+++ b/mobile/android/base/java/org/mozilla/gecko/preferences/GeckoPreferences.java	Thu Aug 03 00:00:00 2017 -0000
@@ -1199,13 +1199,6 @@
             UpdateServiceHelper.setAutoDownloadPolicy(this, UpdateService.AutoDownloadPolicy.get((String) newValue));
         } else if (PREFS_UPDATER_URL.equals(prefName)) {
             UpdateServiceHelper.setUpdateUrl(this, (String) newValue);
-        } else if (PREFS_HEALTHREPORT_UPLOAD_ENABLED.equals(prefName)) {
-            final Boolean newBooleanValue = (Boolean) newValue;
-            AdjustConstants.getAdjustHelper().setEnabled(newBooleanValue);
-            // If Health Report has been disabled Mma should also be stopped.
-            // If it was just enabled, we should also try to start Mma immediately
-            // provided that all the other requirements to start Mma are met.
-            informMmaStatusChanged(newBooleanValue);
         } else if (PREFS_GEO_REPORTING.equals(prefName)) {
             if ((Boolean) newValue) {
                 enableStumbler((CheckBoxPreference) preference);
